### ISSUE TITLE
SelectionArea pops up incorrect context menu when semantics is on

### ISSUE THREAD

**Issue by yjbanov** [flutter/flutter#135318]
The context menu for selected content provided by the `SelectionArea` should provide selection-specific menu options, but when semantics is on it pops up a generic menu:

<img width="355" alt="Screenshot 2023-09-22 at 10 35 00 AM" src="https://github.com/flutter/flutter/assets/211513/cb9a1ae0-db2e-4b6b-98c0-22ee8a63759b">

Repro

```
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

const Color darkBlue = Color.fromARGB(255, 18, 32, 47);

void main() {
  runApp(MyApp());
  SemanticsBinding.instance.ensureSemantics();
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData.dark().copyWith(
        scaffoldBackgroundColor: darkBlue,
      ),
      debugShowCheckedModeBanner: false,
      home: Scaffold(
          body: Center(
            child: MyWidget(),
          ),
        ),
    );
  }
}

class MyWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return SelectionArea(child: Text(
      'Select this text',
      textAlign: TextAlign.center,
      style: Theme.of(context).textTheme.headlineSmall,
    ));
  }
}
```

**Comment by chunhtai** [comment]
behind the theme the selection area create a div to intercept the right click and provide selected text. it may be conflicting with the semantics dom

**Comment by yjbanov** [comment]
Yeah, I'm guessing the semantic DOM sits on top of that div in hit test order and intercepts the right-click. Is the `<div>` rendered as a platform view? Normally, platform view semantics is embedded into the semantic DOM, but maybe what's happening is we have multiple semantics branches for the same content. We're probably rendering both semantics for the `Text` widgets and semantics for the platform view on top of each other.

**Comment by Renzo-Olivares** [comment]
I cannot seem to reproduce this on the latest master commit

<img width="719" height="614" alt="Image" src="https://github.com/user-attachments/assets/06819819-ce6b-43d1-a52c-5ea20cfeba56" />

Is this something you can still reproduce @yjbanov?
